error: AsSuperVisitor: type is not an erased subtype of supertype.

Upgraded checker framework to 3.46.0. Seeing lots of failures with detail except one. I have temporarily suppressed other errors.
There is no reference to it anywhere in the log. Tried with  -X. Still Nothing.
How do figure out where the error is coming from?

```
[INFO] 100 warnings 
[INFO] -------------------------------------------------------------
[INFO] -------------------------------------------------------------
[ERROR] COMPILATION ERROR : 
[INFO] -------------------------------------------------------------
[ERROR] error: AsSuperVisitor: type is not an erased subtype of supertype.
[INFO] 1 error
[INFO] -------------------------------------------------------------
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  02:03 min
[INFO] Finished at: 2024-08-09T14:07:24+05:30
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.10.1:compile (default-compile) on project config-ms: Compilation failure
[ERROR] error: AsSuperVisitor: type is not an erased subtype of supertype.
[ERROR] 
[ERROR] -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException
```

--- a/framework/src/main/java/org/checkerframework/framework/util/typeinference8/constraint/Expression.java
+++ b/framework/src/main/java/org/checkerframework/framework/util/typeinference8/constraint/Expression.java
@@ -280,7 +280,9 @@ public class Expression extends TypeConstraint {
         AbstractType gi = gs.get(i);
         constraintSet.add(new Typing(fi, gi, TypeConstraint.Kind.TYPE_EQUALITY));
       }
-      if (tPrime != T) {
+      @SuppressWarnings("interning:not.interned") // checking for exact object.
+      boolean tPrimeNotSameAsT = tPrime != T;
+      if (tPrimeNotSameAsT) {
         constraintSet.add(new Typing(tPrime, T, TypeConstraint.Kind.SUBTYPE));
       }
     } else {

--- a/framework/src/main/java/org/checkerframework/framework/util/typeinference8/constraint/Expression.java
+++ b/framework/src/main/java/org/checkerframework/framework/util/typeinference8/constraint/Expression.java
@@ -280,7 +280,9 @@ public class Expression extends TypeConstraint {
         AbstractType gi = gs.get(i);
         constraintSet.add(new Typing(fi, gi, TypeConstraint.Kind.TYPE_EQUALITY));
       }
-      constraintSet.add(new Typing(tPrime, T, TypeConstraint.Kind.SUBTYPE));
+      if (tPrime != T) {
+        constraintSet.add(new Typing(tPrime, T, TypeConstraint.Kind.SUBTYPE));
+      }
     } else {
       context.addLambdaParms(lambda.getParameters());
     }

--- a/framework/src/main/java/org/checkerframework/framework/util/typeinference8/types/VariableBounds.java
+++ b/framework/src/main/java/org/checkerframework/framework/util/typeinference8/types/VariableBounds.java
@@ -558,7 +558,7 @@ public class VariableBounds {
   public boolean hasLowerBoundDifferentParam() {
     List<AbstractType> parameteredTypes = new ArrayList<>();
     for (AbstractType type : bounds.get(BoundKind.LOWER)) {
-      if (!type.isUseOfVariable() && type.isParameterizedType()) {
+      if (type.isProper() && type.isParameterizedType()) {
         parameteredTypes.add(type);
       }
     }

--- a/framework/tests/all-systems/Issue6753.java
+++ b/framework/tests/all-systems/Issue6753.java
@@ -1,0 +1,23 @@
+package typearginfer;
+
+import java.util.Set;
+import java.util.concurrent.Callable;
+
+public abstract class Issue6753 {
+
+  abstract <E> E call(Callable<E> task);
+
+  abstract <T> Set<T> list();
+
+  void method(int length) {
+    Set<String> resources =
+        call(
+            () -> {
+              if (length > 1) {
+                return list();
+              } else {
+                return list();
+              }
+            });
+  }
+}

